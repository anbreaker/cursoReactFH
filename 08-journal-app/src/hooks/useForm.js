import {useState} from 'react';

export const useForm = (initialStateForm = {}) => {
  const [formValues, setValues] = useState(initialStateForm);

  const reset = (newFormState = initialStateForm) => {
    setValues(newFormState);
  };

  const handleInputChange = ({target}) => {
    setValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  return [formValues, handleInputChange, reset];
};
