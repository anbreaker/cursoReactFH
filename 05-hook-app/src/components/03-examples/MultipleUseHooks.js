import React from 'react';
import {useFetch} from '../../hooks/useFetch';
import {useCounter} from '../../hooks/useCounter';

export const MultipleUseHooks = () => {
  const {counter, increment} = useCounter(1);

  const {loading, data} = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );
  // console.log(counter);

  const {author, quote} = !!data && data[0];

  return (
    <div className="container">
      <h1>BreakingBad Quotes</h1>
      <hr />

      {loading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="blockquote text-center">
          <p className="mb-4"> {author} </p>
          <footer className="blockquote-footer"> {quote} </footer>
        </blockquote>
      )}
      <button className="bnt btn-primary" onClick={increment}>
        Next Quote
      </button>
    </div>
  );
};
