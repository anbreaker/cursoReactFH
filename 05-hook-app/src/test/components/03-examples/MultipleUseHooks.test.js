import {shallow} from 'enzyme';
import {MultipleUseHooks} from '../../../components/03-examples/MultipleUseHooks';
import {useCounter} from '../../../hooks/useCounter';
import {useFetch} from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en MultipleUseHooks', () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
    });
  });

  test('debe mostarse correctamente', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });

    const wrapper = shallow(<MultipleUseHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar la informacion', () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: 'anbreaker',
          quote: 'Hola Mundo',
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleUseHooks />);

    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-4').text().trim()).toBe('anbreaker');
    expect(wrapper.find('footer').text().trim()).toBe('Hola Mundo');
  });
});
