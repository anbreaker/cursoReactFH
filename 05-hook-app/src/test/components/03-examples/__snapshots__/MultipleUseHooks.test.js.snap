// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleUseHooks debe mostarse correctamente 1`] = `
<div
  className="container"
>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="bnt btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
